client/gutenberg/extensions/likes-and-sharing/panel.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import get from 'lodash/get';
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { CheckboxControl, PanelBody } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+
+class LikesAndSharingPanel extends Component {
+	render() {
+		const { areLikesEnabled, isSharingEnabled, toggleLikes, toggleSharing } = this.props;
+
+		return (
+			<PanelBody title={ __( 'Likes and Sharing' ) }>
+				<CheckboxControl
+					label={ __( 'Show likes.' ) }
+					checked={ areLikesEnabled }
+					onChange={ value => {
+						toggleLikes( value );
+					} }
+				/>
+
+				<CheckboxControl
+					label={ __( 'Show sharing buttons.' ) }
+					checked={ isSharingEnabled }
+					onChange={ value => {
+						toggleSharing( value );
+					} }
+				/>
+			</PanelBody>
+		);
+	}
+}
+
+// Fetch the post meta.
+const applyWithSelect = withSelect( select => {
+	const { getEditedPostAttribute } = select( 'core/editor' );
+	const meta = getEditedPostAttribute( 'meta' );
+
+	return {
+		areLikesEnabled: ! get( meta, [ 'switch_like_status' ] ), // todo site option
+		isSharingEnabled: ! get( meta, [ 'sharing_disabled' ], '' ),
+	};
+} );
+
+// Provide method to update post meta.
+const applyWithDispatch = withDispatch( ( dispatch, { meta } ) => {
+	const { editPost } = dispatch( 'core/editor' );
+
+	// todo: handle switch_like_status logic
+
+	return {
+		toggleLikes( value ) {
+			editPost( { meta: { ...meta, switch_like_status: ! value } } );
+		},
+		toggleSharing( value ) {
+			// const value = get( meta, [ 'sharing_disabled' ] ); -- doesn't work - always false
```

Comment:
I think this is not working because `meta` is not a property of the component. It should work if you set it on `applyWithSelect`. But you can also use directly the `areLikesEnabled` and `isSharingEnabled` properties, instead of getting them from `meta`:

```js
const applyWithSelect = withSelect( select => {
	const { getEditedPostAttribute } = select( 'core/editor' );
	const meta = getEditedPostAttribute( 'meta' );

	return {
		areLikesEnabled: ! get( meta, [ 'switch_like_status' ] ), // todo site option
		isSharingEnabled: ! get( meta, [ 'sharing_disabled' ], '' ),
		meta,
	};
} );

const applyWithDispatch = withDispatch( ( dispatch, { meta, areLikesEnabled, isSharingEnabled } ) => {
	const { editPost } = dispatch( 'core/editor' );

	return {
		toggleLikes: () => editPost( { meta: { ...meta, switch_like_status: ! areLikesEnabled } } ),
		toggleSharing: () => editPost( { meta: { ...meta, sharing_disabled: ! isSharingEnabled } } ),
	}
} );
```

Note that as we're using now `areLikesEnabled` and `isSharingEnabled`  for toggling the values, we don't need to pass any value to `toggleLikes` or `toggleSharing`:

```jsx
<CheckboxControl
	label={ __( 'Show likes.' ) }
	checked={ areLikesEnabled }
	onChange={ toggleLikes }
/>

<CheckboxControl
	label={ __( 'Show sharing buttons.' ) }
	checked={ isSharingEnabled }
	onChange={ toggleSharing }
/>
```

Reply:
Nice simplification, thank you @mmtr!

Reply:
After introducing a bug, and some discussion with @dmsnell, I've un-simplified that simplification to stay explicit about what value we're setting (and just in case something gets out of sync, we're still paying attention to the checkbox value, which is to say, what the user _intends_ to happen).